### DISCUSSION TITLE
"indent" rule doesn't seem to indent operators and operands in new line

### DISCUSSION THREAD

**Discussion by ftzi** [eslint/eslint#14850] [Q&A]
I ain't certain if this is a bug or a config that I am missing out, but:

```js
/*eslint indent: ["warn", 2] */
let a =
3; // No indent
a
+= 4; // No indent
a =
9 + 2; // No indent
```

It doesn't complain if I manually add 2 spaces (or any number) at the beginning of those lines.

Any way to force to be:

```js
let a =
  3;
a
  += 4;
a =
  9 + 2;
```

?

I am aware of operator-linebreak rule, but it doesn't do what I want, that is the spaces at the beginning of the line.

**Comment by nzakas** [comment]
This is beyond the scope of the indent rule, which doesn’t really look at line breaks of individual statements. If you want more fine grained control then you’ll need to create a custom rule.

**Reply by rhnatiuk** [reply]
@nzakas, but this is not making much sense, as it is not consistent at all. eslint indents function declarations, function calls, classes, commas, etc., but then it is not indenting conceptually similar case with operators. For example, the following if statement:

```js
if(someFunction(
a, b, c,
                                      d, e, f,
                g, h, i
) {
```

will be correctly indented as

```js
if(someFunction(
   a, b, c,
   d, e, f,
   g, h, i
) {
```

but very similar statement (there is now + between c and d, and && between f and g)

```js
if(someFunction(
a, b, c
                                      + d, e, f
                && g, h, i
) {
```

will be indented to

```js
if(someFunction(
    a, b, c
                                      + d, e, f
                && g, h, i
) {
```

And this is related to all things - function calls, if statements with && and ||, etc. Mess.

**Reply by nzakas** [reply]
Indentation is very difficult to get correct. The operator makes it a different type of statement and follows different rules. It might be a bug. Feel free to open an issue.
